onramp_compassion/models/sponsorship_correspondence.py

```diff
@@ -93,6 +93,21 @@ def _commkit_update(self, data, message_id=None):
         SponsorshipCorrespondence object. """
         letter_mapping = mapping.new_onramp_mapping(self._name, self.env)
         commkit_vals = letter_mapping.get_vals_from_connect(data)
+
+        # Download and store letter

```

Comment:
You should only download letter image of Published letters, because we can receive notifications before the letter is translated. So you can put your code inside :
```
if commkit_vals.get('state') == 'Published to Global Partner':
```